nucypher/blockchain/eth/decorators.py

```diff
@@ -84,8 +89,37 @@ def wrapped(actor=None, *args, **kwargs):
 
 def save_receipt(actor_method):
     """Decorator to save the receipts of transmitted transactions from actor methods"""
+    @functools.wraps(actor_method)
     def wrapped(self, *args, **kwargs):
         receipt = actor_method(self, *args, **kwargs)
         self._saved_receipts.append((datetime.utcnow(), receipt))
         return receipt
     return wrapped
+
+
+#
+# Contract Function Handling
+#
+
+# TODO: Disable collection in prod (detect test package)
+COLLECT_CONTRACT_API = True
+
+
+class ContractInterfaces(Enum):
+    CALL = 0
+    TRANSACTION = 1
+    ATTRIBUTE = 2
+    UNKNOWN = 3
```

Comment:
:heavy_check_mark: Looks beautiful with constant sorrow! :-)

Reply:
As a follow up - I wonder if this is a good example for documentation in the constant sorrow readme.

Reply:
If would argue that an enum has its advantages when it comes to type checking/namespace pollution. Would be nice if there was a `ConstantSorrow` enum that would allow one to just write

```
class ContractInterfaces(CS.Enum):
    CALL
    TRANSACTION
    ATTRIBUTE
    UNKNOWN
```

and the field got assigned CS constants as values. But I don't want to ask Kieran to redo it once again.

Reply:
Interesting... @jMyles might be interested in this idea.

Reply:
After some thought, I don't think CS is the best place for it. The attributes are not created ad-hoc in each module which is CS's modus operandi, but declared once. And they need to have `ContractInterfaces` type, for consistency. The only CS constant's property that is needed here is that printing an attribute gives you its full name (plus the enum's name, preferably). And the snippet above is actually syntactically incorrect, attributes must have either values or annotations, so it'll look like:
```
class ContractInterfaces(MyEnum):
    CALL: 'ContractInterfaces'
    TRANSACTION: 'ContractInterfaces'
    ATTRIBUTE: 'ContractInterfaces'
    UNKNOWN: 'ContractInterfaces'
```
(which will also help `mypy`). 

All in all, it seems easier to just use the standard library enum and assign CS constants to its attributes...